Feat/get active proposals first

dcrescimbeni opened the pull request:
# Description

The main view for guilds is the "Governance" page. There, it shows "Active" and "Executable" proposals.

To access all proposals, go to "All proposals", on the left sidebar.

Closes #170 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Manual testing.

# Checklist:

- [x] My code follows the existing style of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any UI changes have been tested and made responsive for mobile views

rossneilson reviewed (changes requested):
Looks great Dino! the only thing I think we need is the create proposal button on the governance page as well

dcrescimbeni commented:
I split pages components in their own folders, added `.styled` and `index` files.

This sets up the folder structure to add tests in the future and go in line with our current component structure.

MiltonTulli commented on src/locales/en/translation.json:
```diff
@@ -8,6 +8,7 @@
   "proposals_zero": "No proposals",
   "proposals_one": "{{count}} proposal",
   "proposals_other": "{{count}} proposals",
+  "proposals_all": "All proposals",
```
I think we should call this `allProposals`. 
the `_one`, `_other` are for special use with `count` prop. (also `_zero` for `count:0`). But if you will use this as a normal key (like you are doing in /GuildSidebar.tsx `{t('proposals_all')}`) it should be `allProposals`.
Seee [i18next plurals doc](https://www.i18next.com/translation-function/plurals)

dcrescimbeni replied:
Totally, makes more sense

MiltonTulli commented on src/Modules/Guilds/pages/AllProposals/AllProposals.tsx:
```diff
@@ -0,0 +1,100 @@
+    return clone.reverse();
+  }, [proposalIds]);
+
+  const PROPOSALS_TO_LOAD = 10;
```
Can you define this constant outside your FunctionalComponent?

dcrescimbeni replied:
I can, but where will it make sense?

I extracted a constant as a way of "naming" an arbitrary value of number of proposals to load. This as well could be 5, 15 or 20. But this is only relevant in this context.

MiltonTulli replied:
Outside your component definition is ok for now I think. Just to avoid creating the same const on each render.

MiltonTulli commented on src/Modules/Guilds/pages/AllProposals/AllProposals.tsx:
```diff
@@ -0,0 +1,100 @@
+    return (
+      <Result
+        state={ResultState.ERROR}
+        title="We ran into an error."
```
Can you add translation?

dcrescimbeni replied:
Yep, missed it!

MiltonTulli commented on src/Modules/Guilds/pages/Governance/Governance.tsx:
```diff
@@ -0,0 +1,152 @@
+    return (
+      <Result
+        state={ResultState.ERROR}
+        title="We ran into an error."
```
Translation here too? 😀

MiltonTulli commented on src/Modules/Guilds/pages/Governance/Governance.tsx:
```diff
@@ -0,0 +1,152 @@
+      <ProposalsList data-testid="proposals-list">
+        <StyledHeading size={2}>{t('proposals')}</StyledHeading>
+
+        {activeProposals && activeProposals._hex === '0x00' && (
```
Question.
Is there a scenario where we need to prepare for `activeProposals` without `_hex`? Make sense to do `activeProposals?._hex ...`

dcrescimbeni replied:
Hmmm I don't think I follow. The first `activeProposals` check is to prevent the `undefined` value to throw an error when trying to access the `_hex` property.

Filipv95 commented:
@rossneilson I've also checked this PR and it looks good on FE.
